### PR TITLE
Import prod script

### PR DESCRIPTION
**Pull request by EGCGDanForth**
New script for assigning samples from Sample Review to Data Trigger Release. New script for sending email notification to the Facility Manager to review samples in the Data Trigger Release step. Update to next_step_assignment.py so that "remove" is now an option for the next_step.

### CODE REVIEW

**Comment by EGCGDanForth** [comment]
Hi,

All test scripts added.

Is this in a state that you can review now or is their something else I need to do?

Dan

**Comment by EGCGDanForth** [comment]
I can't resolve the conflict in test_email.py. Will need some help performing a manual merge. The correct version is from my branch import_prod_script, the master version will fail as the UDF names in the LIMS have changed.

**Comment by EGCGDanForth** [comment]
Finally...all conflicts and test fails addressed. Please can someone review.

File: scripts/assign_sample_review.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        stage = get_workflow_stage(self.lims, "PostSeqLab EG 1.0 WF",
+                                   "Data Release Trigger EG 1.0 ST")
+
+        for art in self.output_artifacts:
+
+            if art.udf.get("SR Useable") == "yes":
+                sample=art.samples[0]
+                artifact_to_route.add(sample.artifact)
+
+                self.lims.route_artifacts(list(artifact_to_route
```
Comment (mwhamgenomics) [inline scripts/assign_sample_review.py]: Don't we want this outside the loop so we call `route_artifacts` once on evverything?

File: tests/test_assign_to_workflow_sample_review.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
+from scripts.assign_sample_review import AssignWorkflowSampleReview
+from tests.test_common import TestEPP, fake_artifact
+from unittest.mock import Mock, patch, PropertyMock
+
+
+def fake_all_outputs(unique=False, resolve=False):
+    '''Return a list of mocked artifacts which contain sample which contain artifact ... Simple!'''
+    return (
+        Mock(samples=[Mock(artifact=fake_artifact('a1'), id='s1')], udf={"SR Useable": 'yes'}),
+        Mock(samples=[Mock(artifact=fake_artifact('a2'), id='s2')],udf={"SR Useable":'yes'})
```
Comment (mwhamgenomics) [inline tests/test_assign_to_workflow_sample_review.py]: We could add an artifact marked as `'SR Useable': 'no'` to make sure that non-useable samples don't get routed to data release

File: tests/test_emails.py [hunk 3]
```diff
@@ -233,3 +234,45 @@ def test_send_email(self):
...
+        self.epp = self.create_epp(DataReleaseFMEmail)
+
+    def test_only_one_project(self):
+        try:
+            with self.assertRaises(ValueError):
+                with self.patch_project_multi:
+                    self.epp._run()
+
+        except NotImplementedError:
+            print('Skipping test for abstract class: ' + self.epp.__class__.__name__)
```
Comment (mwhamgenomics) [inline tests/test_emails.py]: This is exactly the same method as in the superclass, so can be removed. `TestDataReleaseFacilityManager` will inherit the method from the superclass and run the test

**Comment by EGCGDanForth** [comment]
Changes requested by mwhamgenomics have been made

**Comment by EGCGDanForth** [comment]
2 new scripts and their tests added. 1 script for checking that step UDFs have been completed - it will be linked in the EPP with && with other scripts so they can only run if the check exits with a 0. 1 script is just for assigning submitted samples to the invoice workflow. As usual, my tests are probably a bit ropey, particular the the check_UDFs test.

**Comment by EGCGDanForth** [comment]
Don't review just yet

**Comment by EGCGDanForth** [comment]
I found developing a test for the sys exit part of the test_check_step_UDF problematic. If you have a solution then let me know.

File: tests/test_check_step_UDFs.py [hunk 4]
```diff
@@ -24,7 +23,21 @@ def setUp(self):
...
+            self.log_file
+        )
+
     def test_check_step_UDFs(self):
         with self.patched_process:
-            self.epp._run()
+            # Both UDFs are present so run does not execute sys.exit
+            assert self.epp._run() is None
 
+    def test_check_step_UDFs(self):
```
Comment (katieemelianova) [inline tests/test_check_step_UDFs.py]: Is this supposed to have the same name as the test above it?

**Pull request merged**